src/lib/storage/value_segment.cpp

```diff
@@ -99,15 +99,19 @@ pmr_vector<T>& ValueSegment<T>::values() {
   return _values;
 }
 
+template <typename T>
+bool ValueSegment<T>::contains_nulls() const {
+  return is_nullable();
```

Comment:
I don't like that. As value segments can be written any time, answering that question is not trivial.
Also, asking the segment if it contains NULLs is something else than a column/segment being nullable.

Reply:
What about `has_null_values_vector()` for all segments?
Or just revert the changes and life with a larger IsNullScan?

Reply:
I reverted them.